apps/comments-ui/src/components/content-box.tsx

```diff
@@ -8,17 +8,36 @@ type Props = {
     done: boolean
 };
 const ContentBox: React.FC<Props> = ({done}) => {
-    const luminance = (r: number, g: number, b: number) => {
-        const a = [r, g, b].map(function (v) {
-            v /= 255;
-            return v <= 0.03928 ? v / 12.92 : Math.pow((v + 0.055) / 1.055, 2.4);
-        });
-        return a[0] * 0.2126 + a[1] * 0.7152 + a[2] * 0.0722;
+    const luminance = (colorString: string) => {
+        if (colorString.startsWith('oklab') || colorString.startsWith('oklch')) {
+            const regexMatches = colorString.match(/ok(?:lab|lch)\(([^ ]+)/);
+            if (!regexMatches || regexMatches.length < 2) {
+                return 0;
+            }
+            const lum = regexMatches[1];
+            if (lum === 'none') {
+                return 0;
+            } else {
+                return parseFloat(lum);
+            }
+        } else if (colorString.startsWith('rgb')) {
+            const colorsOnly = colorString.substring(colorString.indexOf('(') + 1, colorString.lastIndexOf(')')).split(/,\s*/);
+            const r = parseInt(colorsOnly[0]);
+            const g = parseInt(colorsOnly[1]);
+            const b = parseInt(colorsOnly[2]);
+
+            const a = [r, g, b].map(function (v) {
+                v /= 255;
+                return v <= 0.03928 ? v / 12.92 : Math.pow((v + 0.055) / 1.055, 2.4);
+            });
+            return a[0] * 0.2126 + a[1] * 0.7152 + a[2] * 0.0722;
```

Comment:
Same as above -- could be a named helper, would probably make it easier to read.